zerver/views/streams.py

```
@@ -738,14 +740,15 @@ def add_subscriptions_backend(
     result["subscribed"] = dict(result["subscribed"])
     result["already_subscribed"] = dict(result["already_subscribed"])
 
-    send_messages_for_new_subscribers(
-        user_profile=user_profile,
-        subscribers=subscribers,
-        new_subscriptions=result["subscribed"],
-        id_to_user_profile=id_to_user_profile,
-        created_streams=created_streams,
-        announce=announce,
-    )
+    if send_notification_to_new_users and len(result["subscribed"]) <= MAX_SUBS_FOR_NOTIFICATION:
```

I am not sure but should we raise an error if user passes `send_notification_to_new_users` as `True` when adding more than 100 users.

---

Another option is to notify in the Response that notifications weren't sent.